Fix for S-Quadra Advisory #2003-11-26
(http://www.s-quadra.com/advisories/Adv-20031126.txt)
taken from freeradius CVS - change with comment:

 RADIUS attributes can be up to ~256 bytes long.
 This is the pam_smb vulnerability from a while ago...

--- freeradius-0.9.3/src/modules/rlm_smb/smblib.c.orig	2002-08-06 18:50:33.000000000 +0200
+++ freeradius-0.9.3/src/modules/rlm_smb/smblib.c	2003-11-28 20:38:18.699957008 +0100
@@ -316,7 +316,7 @@
 
 { struct RFCNB_Pkt *pkt;
   int param_len, i, pkt_len, pass_len,a;
-  char *p, pword[128];
+  char *p, pword[256];
 
   /* First we need a packet etc ... but we need to know what protocol has  */
   /* been negotiated to figure out if we can do it and what SMB format to  */
